scipy/interpolate/interpnd.pyx

```diff
@@ -909,37 +965,56 @@ class CloughTocher2DInterpolator(NDInterpolatorBase):
 
     def __init__(self, points, values, fill_value=np.nan,
                  tol=1e-6, maxiter=400, rescale=False):
+        self.tol = tol
+        self.maxiter = maxiter
         NDInterpolatorBase.__init__(self, points, values, ndim=2,
-                                    fill_value=fill_value, rescale=rescale)
-        if self.tri is None:
-            self.tri = qhull.Delaunay(self.points)
-        self.grad = estimate_gradients_2d_global(self.tri, self.values,
-                                                 tol=tol, maxiter=maxiter)
+                                    fill_value=fill_value, rescale=rescale,
+                                    need_values=False)
+    
+    def _set_values(self, values, fill_value=np.nan, need_contiguous=True):
+        """
+        Sets the values of the interpolation points.
 
-    def _evaluate_double(self, xi):
+        Parameters
+        ----------
+        values : ndarray of float or complex, shape (npoints, ...)
+            Data values.
+        """
+        NDInterpolatorBase._set_values(self, values, fill_value=fill_value, need_contiguous=need_contiguous)
+        if self.values is not None:
+            self.grad = estimate_gradients_2d_global(self.tri, self.values,
+                                                    tol=self.tol, maxiter=self.maxiter)
+    
+    def _calculate_triangulation(self, points):
+        self.tri = qhull.Delaunay(points)
+
+    def _evaluate_double(self, xi=None):
+        if xi is None:
+            xi = self._xi
         return self._do_evaluate(xi, 1.0)
 
-    def _evaluate_complex(self, xi):
+    def _evaluate_complex(self, xi=None):
+        if xi is None:
+            xi = self._xi
         return self._do_evaluate(xi, 1.0j)
-
+        
```

Comment:
Nit: trailing whitespace is typically frowned upon.

Reply:
Removed